### PR TITLE
80838  Delete saved filter

### PR DESCRIPTION
**Pull request by bratli**
DevOps#80838

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 1]
```diff
@@ -57,6 +57,16 @@ public void AddSavedFilter(SavedFilter savedFilter)
             _savedFilters.Add(savedFilter);
         }
 
+        public void RemoveSavedFilter(SavedFilter savedFilter)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs:60]: misses test in PersonTests. Same for AddSavedFilter above

File: src/Equinor.ProCoSys.IPO.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs [hunk 2]
```diff
@@ -34,5 +34,10 @@ public class SavedFilterValidator : ISavedFilterValidator
                       && s.ProjectName == projectName
                 select s).AnyAsync(token);
         }
+
+        public async Task<bool> ExistsAsync(int savedFilterId, CancellationToken token) =>
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs:38]: misses test in SavedFilterValidatorTests

File: src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/PersonRepository.cs [hunk 3]
```diff
@@ -17,5 +17,8 @@ public Task<Person> GetWithSavedFiltersByOidAsync(Guid oid)
             => DefaultQuery
                 .Include(p => p.SavedFilters)
                 .SingleOrDefaultAsync(p => p.Oid == oid);
+
+        public void RemoveSavedFilter(SavedFilter savedFilter)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/PersonRepository.cs:21]: This class has no tests

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Persons/PersonsControllerNegativeTests.cs [hunk 4]
```diff
@@ -31,13 +31,43 @@ public async Task CreateSavedFilter_AsHacker_ShouldReturnBadRequest_WhenUnknownP
...
         #endregion
+
+        #region DeleteSavedFilter
+        [TestMethod]
+        public async Task DeleteSavedFilter_AsAnonymous_ShouldReturnUnauthorized()
+            => await PersonsControllerTestsHelper.DeleteSavedFilterAsync(
+                UserType.Anonymous,
+                TestFactory.UnknownPlant,
+                1,
+                "rowVersion",
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Persons/PersonsControllerNegativeTests.cs:50]: Use a valid rowVersion (so an invalid rowVersion do not mess up the issue under test). Same for tests below

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Persons/PersonsControllerTests.cs [hunk 5]
```diff
@@ -39,21 +38,21 @@ public async Task CreateSavedFilter_AsViewer_ShouldSaveFilter()
         public async Task GetSavedFiltersInProject_AsViewer_ShouldGetFilters()
         {
             // Act
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Persons/PersonsControllerTests.cs:40]: This Act-line is misplaced

**Review by janid1967 — APPROVED** [review]
Approved w suggestions

**Review by janid1967 — CHANGES_REQUESTED** [review]
fix tests. Fails in pipeline

File: src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/PersonRepository.cs (hunk 3 above)
Comment (bratli) [reply src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/PersonRepository.cs:21]: True. Adding now. Guess I should make a TECH to add tests in preservation as well? (no PersonRepositoryTests.cs in preservation)

**Review by janid1967 — APPROVED** [review]

**Pull request merged**